src/gt4py/next/type_system_2/inference.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
```

Comment:
probably handling annotations should use the eve utilities, because there are often changes between Python versions and we need to canonicalize the representation